File: research/explorations/bench-realistic/strings.py
import sys, time

def bench(n):
    s = 0
    for i in range(n):
        t = "item-" + str(i)
        parts = t.split("-")
        j = "_".join(parts)
        s += len(j)
    return s

N = int(sys.argv[1]) if len(sys.argv) > 1 else 1000
# warmup
for _ in range(100):
    bench(N)
iters = 10000
start = time.monotonic_ns()
for _ in range(iters):
    r = bench(N)
elapsed = time.monotonic_ns() - start
print(f"result:     {r}")
print(f"iterations: {iters}")
print(f"total:      {elapsed / 1e6:.2f}ms")
print(f"per call:   {elapsed // iters}ns")

File: research/explorations/bench-realistic/strings.js
function bench(n) {
    let s = 0;
    for (let i = 0; i < n; i++) {
        const t = "item-" + String(i);
        const parts = t.split("-");
        const j = parts.join("_");
        s += j.length;
    }
    return s;
}

const N = parseInt(process.argv[2]) || 1000;
// warmup
for (let i = 0; i < 1000; i++) bench(N);
const iters = 10000;
const start = process.hrtime.bigint();
let r;
for (let i = 0; i < iters; i++) r = bench(N);
const elapsed = Number(process.hrtime.bigint() - start);
console.log(`result:     ${r}`);
console.log(`iterations: ${iters}`);
console.log(`total:      ${(elapsed / 1e6).toFixed(2)}ms`);
console.log(`per call:   ${Math.floor(elapsed / iters)}ns`);
